domain-cc/cc-app/src/python_src/api.py

```diff
@@ -47,35 +50,51 @@ def get_health_status():
     return {"status": "ok"}
 
 
+def log_lookup_table_match(
+    classification_code: int, is_in_dropdown: bool, contention_text: str
+):
+    log_contention_text = contention_text if is_in_dropdown else "Not in dropdown"
+
+    if classification_code:
+        already_mapped_text = (  # being explicit, do not leak PII
+            contention_text.strip().lower()
+        )
+        log_as_json({"Lookup table match": already_mapped_text})
+    elif is_in_dropdown:
+        log_as_json({"No lookup table match for dropdown entry": log_contention_text})
+    else:
+        log_as_json("No Lookup table match for free text")
+
+
+def log_as_json(obj: Union[dict, str]):
+    try:
+        obj["date"] = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())
+        obj["level"] = logging.getLevelName(logging.root.level)
```

Comment:
If someone passes in an object with existing `date` or `level` fields, they'll be silently overwritten. May want to leave those fields as-is or raise an error if they already exist - your call.